hack/clean-e2e.sh

```diff
@@ -3,7 +3,6 @@
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 cd "${REPO_ROOT}" || exit 1
 
-minikube delete
```

Comment:
Why don't we delete the kind cluster here anymore?

Reply:
It is cleaned up [here](https://github.com/metal3-io/baremetal-operator/blob/f672c9a8c42607206f427350dcde636f762d1404/test/e2e/e2e_suite_test.go#L323-L331)